go/launchd/launchd.go

```diff
@@ -128,6 +128,28 @@ func (s Service) Restart() error {
 	return Restart(s.Label(), s.log)
 }
 
+// WaitForStatus waits for service status to be available
+func (s Service) WaitForStatus(wait time.Duration) (*ServiceStatus, error) {
+	t := time.Now()
+	i := 1
+	for time.Now().Sub(t) < wait {
+		status, err := s.LoadStatus()
+		if err != nil {
+			return nil, err
+		}
+		if status != nil {
+			return status, nil
+		}
+		// Tell user we're waiting for status after 4 seconds, every 4 seconds
+		if i%4 == 0 {
+			s.info("Waiting for %s to be loaded...", s.label)
+		}
+		time.Sleep(time.Second)
+		i++
+	}
+	return nil, nil
```

Comment:
Then the caller has to check for `nil` and not crash, which I believe isn't the case currently....